### DISCUSSION TITLE
Why Renovate uses latest version on "PNPM" instead of the package.json`s "engines" version?

### DISCUSSION THREAD

**Discussion by tal-rofe** [renovatebot/renovate#22140] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working. Once PNPM upgrades their version, the issue occurs.
I have read this thread: https://github.com/renovatebot/renovate/discussions/12214#discussioncomment-1503089

### Wanted end result.

I'm running Renovate app on GitHub. When Renovate runs, it uses the latest version of `pnpm`, instead using my `engines` version. That causes me issues, because I set `engine-strict="true"` in `.npmrc` file.

You can see the issue here: https://github.com/Exlint/dashboard/pull/1055

### What you tried so far.

I set in the `package.json` file:
```json
	"engines": {
		"pnpm": "8.4.0",
		"node": "18.16.0"
	}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
installing v2 tool node v18.16.0
linking tool node v18.16.0
node: v18.16.0 /usr/local/bin/node
npm: 9.5.1  /usr/local/bin/npm
Installed v2 /usr/local/buildpack/tools/v2/node.sh in 3 seconds
installing v2 tool pnpm v8.5.0
linking tool pnpm v8.5.0
8.5.0
Installed v2 /usr/local/buildpack/tools/v2/pnpm.sh in 2 seconds
 ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

Your pnpm version is incompatible with "/mnt/renovate/gh/Exlint/dashboard".

Expected version: 8.4.0
Got: 8.5.0

This is happening because the package's manifest has an engines.pnpm field specified.
To fix this issue, install the required pnpm version globally.

To install the latest version of pnpm, run "pnpm i -g pnpm".
To check your pnpm version, run "pnpm -v".
```

</details>

**Comment by rarkins** [comment]
@RahulGautamSingh please make sure we have a bug report issue for this open in case you are not able to fix it right away

**Comment by RahulGautamSingh** [comment]
@tal-rofe The PR you linked has been merged now. Is Renovate working as expected now? 
I tested one locally on the lastes renovate commit and it worked fine. Let me know if that is not the case.

**Reply by tal-rofe** [reply]
@RahulGautamSingh 
No, It was merged because there was already an another PR to upgrade pnpm to latest version, therefore this PR was rebased with new pnpm version and ran successfully... I attached the logs anyway

**Reply by tal-rofe** [reply]
@RahulGautamSingh 

Here we are again. I'm using `8.5.0` pnpm version, and now Renovate uses `8.5.1`
https://github.com/Exlint/dashboard/pull/1079

**Reply by RahulGautamSingh** [reply]
Related issue: https://github.com/renovatebot/renovate/issues/22250

**Comment by samydoesit** [comment]
I recently noted that Renovatebot switched to using pnpm as its package manager. However, when running the config-validator of Renovatebot with our local package.json's `engines` constraint for `pnpm`, the validation fails with the following error:

```bash
ERR_PNPM_UNSUPPORTED_ENGINE  Unsupported environment (bad pnpm and/or Node.js version)

This error happened while installing a direct dependency of /tmp/directory/.pnpm-store/v3/tmp/dlx-468
Your pnpm version is incompatible with "registry.npmjs.org/renovate/36.34.0".

Expected version: ^8.6.11
Got: 8.6.5
```

- We are using pnpm version `8.6.5`.
- We have set `engine-strict="true"` in `.npmrc`
- It seems Renovatebot is on pnpm version `8.6.11`.

- Downgrading to Renovate `36.31.0` serves as a temporary workaround for us.

**Reply by rarkins** [reply]
Please file a separate discussion
